src/format/elf64/section.c

```diff
@@ -0,0 +1,296 @@
+{
+    Elf64_Offset offset = 0;
+    offset = header->offset;
+    return offset;
+}
+
+/**
+ * Get the length of an ELF64 section's data.
+ *
+ * @param header The ELF64 section header to read.
+ * @return The length of the section's data, in bytes.
+ */
+extern Elf64_Xword elf64_get_section_size(const ELF64_Section_Header* header)
+{
+    Elf64_Xword length = 0;
+    length = header->offset;
+    return length;
+}
+
+/**
+ * Get an ELF64 section's link table index.
+ *
+ * @param header the ELF64 section header to read.
+ * @return The section's link table index.
+ */
+extern Elf64_Word elf64_get_section_link_table_index(
+    const ELF64_Section_Header* header)
+{
+    Elf64_Word link_index = 0;
+    link_index = header->link;
+    return link_index;
+}
+
+/**
+ * Get an ELF64 section's extra information.
+ *
+ * @param header The ELF64 section header to read.
+ * @return The section's extra information. Semantics are CPU dependent.
+ */
+extern Elf64_Word elf64_get_section_extra_info(
+    const ELF64_Section_Header* header)
+{
+    Elf64_Word info = 0;
+    info = header->info;
+    return info;
+}
+
+/**
+ * Get an ELF64 section's alignment restriction.
+ *
+ * @param header The ELF64 section header to read.
+ * @return The section's mandatory data alignment.
+ */
+extern Elf64_Xword elf64_get_section_alignment(
+    const ELF64_Section_Header* header)
+{
+    Elf64_Xword align = 0;
+    align = header->address_align;
+    return align;
+}
+
+/**
+ * Get an ELF64 section's fixed entry size, if any exists.
+ *
+ * @param header The ELF64 section header to read.
+ * @return The section's fixed entry size.
+ */
+extern Elf64_Xword elf64_get_section_entry_size(
+    const ELF64_Section_Header* header)
+{
+    Elf64_Xword entry_size = 0;
+    entry_size = header->entry_size;
+    return entry_size;
+}
+
+/**
+ * Prints an ELF64 sections data to the standard out.
+ *
+ * @param header The ELF64 section header to print.
+ */
+extern void elf64_print_section_info(const ELF64_Section_Header* header)
+{
+    printf("--- ELF64 Section Header ---\n");
+    printf("ELF64 section type: %s\n",
+        elf64_get_section_type_string(elf64_get_section_type(header)));
+    printf("ELF64 section flags: 0x%lx\n", elf64_get_section_flags(header));
+    printf("ELF64 section load address: 0x%lx\n",
+        elf64_get_section_address(header));
+    printf("ELF64 section offset: 0x%lx\n", elf64_get_section_offset(header));
+    printf("ELF64 section size: 0x%lx\n", elf64_get_section_size(header));
+    printf("ELF64 section link table index: 0x%x\n",
+        elf64_get_section_link_table_index(header));
+    printf("ELF64 section extra info: 0x%x\n",
+        elf64_get_section_extra_info(header));
+    printf("ELF64 section alignment restriction: 0x%lx\n",
+        elf64_get_section_alignment(header));
+    printf("ELF64 section fixed entry size: 0x%lx\n",
+        elf64_get_section_entry_size(header));
+}
```

Comment:
Remove this. It depends on a hosted C std lib, which `libprim` cannot assume.

We could consider moving print functionality to a `platform` dependent object.